searx/engines/tubearchivist.py

```diff
@@ -0,0 +1,173 @@
+  5. Mount this new configuration over the default configuration.
+
+    - With docker run, this would be `-v ./your-new-config.yml:/etc/nginx/sites-available/default`
+    - With docker compose, this would be
+        `- "./your-new-config.yml:/etc/nginx/sites-available/default:ro"`
+
+  6. Start the TA container.
+
+After these steps, double check that TA works as normal (nothing should be different on the TA side).
+Searching again should now show images.
+
+"""
+
+from urllib.parse import urlencode
+from dateutil.parser import parse
+from searx.utils import html_to_text, humanize_number
+from searx.enginelib.traits import EngineTraits
+
+traits: EngineTraits
+
+about = {
+    # pylint: disable=line-too-long
+    "website": 'https://www.tubearchivist.com',
+    "official_api_documentation": 'https://github.com/tubearchivist/tubearchivist',
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+# engine dependent config
+categories = ["videos"]
+paging = True
+
+
+base_url = ""
+"""Base URL of the Tube Archivist instance.  Fill this in with your own tubearchivist url
+"""
+
+
+token = ""
+"""The API key to use for authentication.
+"""
+
+link_to_mp4 = False
+"""Optional, if true SearXNG will link directly to the mp4 of the video to play in the browser.
+The default behavior is to link into TubeArchivist's interface directly.
+"""
+
+
+def request(query, params):
+    """Assemble request for the Tubearchivist API"""
+
+    if not query:
+        return False
+
+    params['url'] = (
+        base_url.rstrip("/")
+        + "/api/search?"
+        + urlencode(
+            {
+                'query': query,
+            }
+        )
+    )
+    params['headers']['Authorization'] = f'Token {token}'
+
+    return params
+
+
+def response(resp):
+    return video_response(resp)
+
+
+def video_response(resp):
+    """Parse video response from Tubearchivist instances."""
+    results = []
+
+    json_data = resp.json()
+
+    if 'results' not in json_data:
+        return []
+
+    for channel_result in json_data['results']['channel_results']:
+        metadata = []
+        channel_url = absolute_url(f'/channel/{channel_result["channel_id"]}')
+        results.append({
+            'url': channel_url,
+            'title': channel_result['channel_name'],
+            'content': html_to_text(channel_result['channel_description']),
+            'author': channel_result['channel_name'],
+            'views': humanize_number(channel_result['channel_subs']),
+            'thumbnail': f'{absolute_url(channel_result["channel_thumb_url"])}?auth={token}',
+            'metadata': ' | '.join(metadata)
+        })
+
+
+    for video_result in json_data['results']['video_results']:
+        metadata = [
+            x
+            for x in [
+                video_result['channel']['channel_name'],
+                ' | '.join(video_result['tags']),
+            ]
+            if x
+        ]
```

Comment:
Metadata should usually not be more than one or two lines, I think we should remove some information from it to make it easier to overview by the users (e.g. only the first 5 items, or drop the tags)

Reply:
Changed to take a max of 5 items